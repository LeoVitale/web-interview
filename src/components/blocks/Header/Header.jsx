import React from 'react';
import MenuItem from 'components/blocks/MenuItem';
import Avatar from 'components/blocks/Avatar';
import styles from  './Header.module.scss';

const Header = ({ logo }) => {
  const { appHeader, appLogo, menu } = styles;
  return (
    <header className={appHeader}>
      <img src={logo} className={appLogo} alt="logo" />
      <nav className={menu}>
        <MenuItem label="Book" to="/new-appointments" />
        <MenuItem label="Appointments" to="/" />
        <MenuItem label="Family Members" to="/family-member" />
        <Avatar />
      </nav>
    </header>
  )
}

export default Header;
