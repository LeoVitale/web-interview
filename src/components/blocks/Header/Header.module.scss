@import 'styles/_constants.scss';

.appHeader {
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: $gutter-size--large;

    .appLogo {
      max-height: 33px;
    }
    border-bottom: $border-standard;

    .menu {
      display: flex;
      align-items: center;
    }
  }
