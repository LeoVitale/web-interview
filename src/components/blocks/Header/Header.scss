@import '../../../constants';

.app-header {
    padding: $gutter-size--large;

    .app-logo {
      max-height: 33px;
    }
    border-bottom: $border-standard;
  }
