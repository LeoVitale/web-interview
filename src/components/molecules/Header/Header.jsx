import React from 'react'
import PropTypes from 'prop-types'
import MenuItem from 'components/atoms/MenuItem'
import Avatar from 'components/atoms/Avatar'
import styles from './Header.module.scss'

const Header = ({ logo }) => {
  const { appHeader, appLogo, menu } = styles
  return (
    <header className={appHeader}>
      <img src={logo} className={appLogo} alt="logo" />
      <nav className={menu}>
        <MenuItem label="Book" to="/new-appointments" />
        <MenuItem label="Appointments" to="/" />
        <MenuItem label="Family Members" to="/family-members" />
        <Avatar size="small"/>
      </nav>
    </header>
  )
}

Header.propTypes = {
  logo: PropTypes.string.isRequired,
}

export default Header
