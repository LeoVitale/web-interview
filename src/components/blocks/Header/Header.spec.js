import React from 'react';
import { mount } from 'enzyme';
import Header from './Header.jsx';
import Index from '.';

describe('<Header />', () => {
  const header = mount(<Header />);

  it('test index.js import', () => {
    expect(Index).toHaveLength(1);
  });

  it('should render Header as expected', () => {
    expect(header).toMatchSnapshot();
  });
});
