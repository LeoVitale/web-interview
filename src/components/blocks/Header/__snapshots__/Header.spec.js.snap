// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render Header as expected 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Header>
      <header
        className="appHeader"
      >
        <img
          alt="logo"
          className="appLogo"
        />
        <nav
          className="menu"
        >
          <MenuItem
            label="Book"
            to="/new-appointments"
          >
            <Link
              className="menuItem"
              replace={false}
              to="/new-appointments"
            >
              <a
                className="menuItem"
                href="/new-appointments"
                onClick={[Function]}
              >
                <i />
                <span>
                  Book
                </span>
              </a>
            </Link>
          </MenuItem>
          <MenuItem
            label="Appointments"
            to="/"
          >
            <Link
              className="menuItem"
              replace={false}
              to="/"
            >
              <a
                className="menuItem"
                href="/"
                onClick={[Function]}
              >
                <i />
                <span>
                  Appointments
                </span>
              </a>
            </Link>
          </MenuItem>
          <MenuItem
            label="Family Members"
            to="/family-member"
          >
            <Link
              className="menuItem"
              replace={false}
              to="/family-member"
            >
              <a
                className="menuItem"
                href="/family-member"
                onClick={[Function]}
              >
                <i />
                <span>
                  Family Members
                </span>
              </a>
            </Link>
          </MenuItem>
          <Avatar>
            <div
              className="avatar"
            >
              <img
                alt=""
                src="https://react.semantic-ui.com/images/avatar/small/elliot.jpg"
              />
            </div>
          </Avatar>
        </nav>
      </header>
    </Header>
  </Router>
</BrowserRouter>
`;
